src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -316,20 +314,27 @@ public KruizeObject prepareRecommendations(int calCount) {
     public void generateRecommendations(KruizeObject kruizeObject) {
 
         for (K8sObject k8sObject : kruizeObject.getKubernetes_objects()) {
-            for (String containerName : k8sObject.getContainerDataMap().keySet()) {
-                ContainerData containerData = k8sObject.getContainerDataMap().get(containerName);
+            if (k8sObject.getExperimentType().equalsIgnoreCase(AnalyzerConstants.ExperimentTypes.NAMESPACE_EXPERIMENT)) {
+                String namespaceName = k8sObject.getNamespace();
+                NamespaceData namespaceData = k8sObject.getNamespaceData();
+                LOGGER.info("Generating recommendations for namespace: {}", namespaceName);
+                generateNamespaceRecommendations(namespaceData, kruizeObject);
```

Comment:
Keep the uniformity in the function name here. Either change this to `generateRecommendationsBasedOnNamespace` or change the container one according to this.

Reply:
I've updated the function name from generatedNamespaceRecommendations to generateRecommendationsBasedOnNamespace to ensure consistency with the existing naming conventions.